src/docfx/publish/PublishModelBuilder.cs

```diff
@@ -23,14 +24,16 @@ internal class PublishModelBuilder
         MonikerProvider monikerProvider,
         BuildOptions buildOptions,
         SourceMap sourceMap,
-        DocumentProvider documentProvider)
+        DocumentProvider documentProvider,
+        ContributionProvider contributionProvider)
     {
         _config = config;
         _errors = errors;
         _monikerProvider = monikerProvider;
         _locale = buildOptions.Locale;
         _sourceMap = sourceMap;
         _documentProvider = documentProvider;
+        _contributionProvider = contributionProvider;
```

Comment:
It seems a bit strange for a publishing model to rely on `ContributionProvider`. Is there any chance to let `SourceMap` to provide the source URL?

Reply:
For example, is it possible to add `ResolveSourceUrl` similar to https://github.com/dotnet/docfx/blob/c9a73bb193cd0808301b3cdc6f41a520b1878751/src/docfx/restore/FileResolver.cs#L104-L104 in `FileResolver`?